modules/local/select_inflation.nf

```diff
@@ -0,0 +1,46 @@
+process SELECT_INFLATION {
+    tag "Selecting best MCL Inflation"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::bioconductor-r-tidyverse==1.2.1" : null)
+    container "${ workflow.containerEngine == 'docker' ?
+        'austinhpatton123/select_inflation_param:latest': '' }"
+        
+    publishDir(
+        path: "${params.outdir}/orthogroup-summaries",
+        mode: 'copy',
+        saveAs: { fn -> fn.substring(fn.lastIndexOf('/')+1) },
+    )
+
+    input:
+    path og_summaries // Files with summaries of orthogroups inferred using a specific inflation parameter
+
+    output:
+    path "best-inflation-param.txt", emit: best_inflation
+    path "inflation_summaries.pdf",  emit: summary_plot
+    path "versions.yml" ,            emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    """
+    # Pull in the summaries produced by each MCL inflation parameter and 
+    # identify the parameter value that produces the best quality results.
+
+    # Run the script to summarize and produce a figure of these results. 
+    Rscript $projectDir/bin/select_inflation.R
+    
+    # And spit out the value of the selected inflation parameter to be 
+    # captured into a channel from stdout 
+    sed -i "s/\\[1] //g" best-inflation-param.txt
+    
+    cat <<- END_VERSIONS > versions.yml
+    "${task.process}":
+        tidyverse: \$( grep "tidyverse" version.txt | sed "s/\\[1] ‘//g" | sed "s/’//g" )
```

Comment:
does this work like how it's written? With other command-line programs it works like this since you call them directly from the command line but I think technically to get R package versions you have to be in an R session?

Reply:
in the R package called by the module I write out the package versions to a .txt file which then I access here:

# And write out package versions
sink("version.txt")
packageVersion("tidyverse")
packageVersion("reshape")
packageVersion("cowplot")
packageVersion("elbow")
sink()